### PR TITLE
refactor(java): delegate tagging to release-please

### PR DESCRIPTION
**Pull request by chingor13**
release PRs from release-please are now getting more complex. release-please should be in charge of doing to actual tagging for a release that it proposed.

This PR switches Java over using release-please for tagging.

### CODE REVIEW

File: autorelease/__main__.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 def _determine_github_token(github_token):
     """Automatically use the GitHub token provided by Keystore if needed."""
     if github_token is not None:
-        return
+        return github_token
```
Comment (bcoe) [inline autorelease/__main__.py:28]: I'm a bit confused as to why were weren't doing this before.

File: releasetool/commands/tag/java.py [hunk 2]
```diff
@@ -130,20 +131,22 @@ def tag(ctx: TagContext = None) -> TagContext:
...
+        [
+            "release-please",
+            f"--token={ctx.token}",
+            f"--default-branch={default_branch}",
+            "--release-type=java-yoshi",
+            "--bump-minor-pre-major=true",
+            f"--repo-url={repo}",
+            "--debug",
+            "--package-name=",
+            "github-release",
```
Comment (bcoe) [inline releasetool/commands/tag/java.py:147]: I'd be tempted to move the positional to the front before the flags (_I vaguely remember there being a bug in yargs related to this).

**Review by bcoe — APPROVED** [review]
a nit and a question 👍

File: autorelease/__main__.py (hunk 1 above)
Comment (chingor13) [reply autorelease/__main__.py:28]: The current execution flow doesn't pass the token from the command line. It's exercising the logic below this line to read from the keystore file location

File: releasetool/commands/tag/java.py (hunk 2 above)
Comment (chingor13) [reply releasetool/commands/tag/java.py]: Done.

**Review by bcoe — APPROVED** [review]

**Pull request merged**